File: Python/BinaryTreePreOrderTraversal.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left
        self.right = right



class Solution:
    def preOrderTraversal(root):

        outputList = []

        def helper(node,outputList):
            if not node:
                return None
            
            outputList.append(node.val)
            helper(node.left,outputList)
            helper(node.right,outputList)


        helper(root,outputList)

        return outputList

File: Javascript/BinaryTreePreOrderTraversal.js
function TreeNode(val,left,right){
    this.val = (val === undefined) ? 0:val
    this.left = (left === undefined) ? null:left
    this.right = (right === undefined) ? null:right
}

var preOrderTraversal = function(root){
    const outputArray = []

    function helper(node){
        if(node === null){
            return null
        }

        outputArray.push(node.val)
        helper(node.left)
        helper(node.right)
    }

    helper(root)

    return outputArray
}